### DISCUSSION TITLE
finding the prominent group

### DISCUSSION THREAD

**Discussion by guyroznb** [networkx/networkx#4549] [Algorithms]
hi,
After merging #4435 PR, there is an algorithm with the same basic concept and by the same authors which finds the group with the highest betweenness centrality (prominent group).
The paper presents two algorithms to find the prominent group - DFBnB method to find the exact group and greedy algorithm to find the approximate group.
Is there any use for such an algorithm in networkx? Should I implement the greedy approximate algorithm?

[https://www.researchgate.net/profile/Rami_Puzis2/publication/220308855_Finding_the_Most_Prominent_Group_in_Complex_Networks/links/0912f5081a757ed735000000.pdf](url)

**Comment by dschult** [comment]
Yes, those would be nice. 
If you name the approximate version something like `greedy_prominent_group_approximation` that will make it clear it is not exact -- especially with description in the docstring as well.  Or, we could move that function to `networkx/algorithms/approximations`.  My inclination is to leave it here and name/document it well. But I'm open to either.

Thanks!

**Reply by guyroznb** [reply]
just to be clear, do you think that I should implement both the approximate version in `networkx/algorithms/approximations` and the exact implementation in `networkx/algorithms`?

**Comment by dschult** [comment]
Sorry for the confusing wording.

I'm open to suggestions, but I was thinking we should keep **the two implementations in the same module with one function for exact solutions and another for approximations**.  

There is a sub-package for approximate algorithms, but most of those are for problems that get very slow very quickly with size.  We also have other approximate algorithms next to their exact siblings mostly for problems that get very slow for moderate or larger size.  I'm thinking this case is in the latter category.  It is also true that wherever we put it can be changed easily later if there is a reason to do that.
